infracheck/checks/domain-expiration

```diff
@@ -47,43 +56,79 @@ class DomainCheck:
         if not self._domain or self._domain == '':
             raise Exception('Domain must be specified')
 
-    def _check_with_wait(self):
+    def _check_with_wait(self) -> WhoisResult:
         time.sleep(1)
         max_retries = 10
         retries = 0
 
-        while True:
+        while retries <= max_retries:
             try:
-                query = self.whois.query(self._domain)
+                query = self._ask_whois_library(self._domain)
 
                 if not query or not query.expiration_date:
                     return self._parse_shell_whois(self._domain)
 
                 return query
+
             except Exception as e:
                 retries += 1
 
                 if retries > max_retries:
                     raise e
 
+                if isinstance(e, whois.exceptions.UnknownTld):
+                    shell_check_result = self._parse_shell_whois(self._domain)
+
+                    if not shell_check_result or not shell_check_result.expiration_date:
```

Comment:
Why `continue` if check doesn't return None?

Reply:
To `retry` in the loop